tests/publiccloud/storage_perf.pm

```diff
@@ -17,6 +17,38 @@ use Mojo::JSON;
 use constant NUMJOBS => 4;
 use constant IODEPTH => 4;
 
+=head2  get_median_from_db
+
+    Calculates the mean value for given C<load_type> , C<scenario> , C<os_flavor> , C<os_version> for certain C<timeframe>.
+    In case getting empty response from InfluxDB will return C<undef> or mean value calculated by InfluxDB.
+
+=cut
+sub get_median_from_db {
+    my (%args) = @_;
+
+    my $query = sprintf("SELECT MEAN(%s) FROM storage WHERE scenario='%s' and os_flavor='%s' and os_version='%s' %s", $args{load_type}, $args{scenario}, $args{os_flavor}, $args{os_version}, $args{timeframe});
+
+    my $json_res = influxdb_read_data($args{url}, $args{db}, $query);
+    unless (defined($json_res->{results}->[0]->{series})) {
+        record_info('Info', sprintf("No results for load_type=%s, scenario=%s, Flavor=%s, Version=%s for timeframe=[%s]\n", $args{load_type}, $args{scenario}, $args{os_flavor}, $args{os_version}, $args{timeframe}));
+        return undef;
+    }
+
+    my $series = $json_res->{results}->[0]->{series};
+    my @values = @{$series}[0]->{values};
+    return $values[0][0][1];
```

Comment:
Can you a bit comment this? Maybe with example json snippet? So one understands what `$values[0][0][1]` mean?

Reply:
+1 to make some comment with example json here.
I would also guard this, so the arrays do really have this content. 
Other wise you are face the trap of https://perldoc.perl.org/perlglossary#autovivification ,but I think in this case it doesn't matter.

```bash
:~> perl -e 'use Mojo::Base -strict; use Data::Dumper; my @foo; say Dumper(\@foo); say ($foo[0][0][1] // "undef"); say Dumper(\@foo);'
$VAR1 = [];

undef
$VAR1 = [
          [
            []
          ]
        ];
```

Reply:
done